tools/generators/ethereum/command.go.tmpl

```diff
@@ -89,18 +89,27 @@ func {{$contract.ShortVar}}{{$method.CapsName}}(c *cobra.Command, args []string)
         return err
     }
 
-   	{{- range $i, $param := .ParamInfos }}
-   	{{$param.Name}}, err := {{$param.ParsingFn}}(args[{{$i}}])
+   	{{ range $i, $param := .ParamInfos }}
+    {{ if $param.Structured }}
+        {{ $param.Name }} := {{ $param.GoType }}{}
+        if err:= json.Unmarshal([]byte(args[{{ $i }}]), &{{ $param.Name }}); err != nil {
+            return fmt.Errorf("failed to unmarshal {{ $param.Name }} to {{ $param.GoType }}: %w", err)
+        }
+    {{- else -}}
+   	{{ $param.Name }}, err := {{ printf "args[%d]" $i | printf $param.ParsingFn }}
    	if err != nil {
 		return fmt.Errorf(
 			"couldn't parse parameter {{$param.Name}}, a {{$param.Type}}, from passed value %v",
 			args[{{$i}}],
 		)
    	}
-   	{{ end }}
+   	{{- end }}
+    {{- end }}
 
     result, err := contract.{{$method.CapsName}}AtBlock(
-        {{ $method.Params -}}
+        {{- range $i, $param := .ParamInfos }}
+        {{ $param.Name }},
+        {{- end }}
```

Comment:
I was thinking about renaming `ParamInfos` to `CmdArgs` as we use them only in the commands template. Would it help?

Reply:
I think it's a good idea but I'd go even further:

```
paramDeclarations += fmt.Sprintf("%v %v,\n", paramName, goType)
params += fmt.Sprintf("%v,\n", paramName)

// Build CmdArgsInfo used for CLI code generator
cmdParamName := paramName
cmdParsingFn := ""
cmdParamStructured := false

if param.Type.TupleType != nil {
  cmdParamName += "_json"
  cmdParamStructured = true
} else {
  // (...)
}
```

1. I would rename `parsingFn` to `cmdParsingFn`.
2. I would rename `paramStructured` to `cmdParamStructured`.
3. I would use a separate `cmdParamName` variable instead of modifying `paramName` value.

This is what I found the most confusing during the initial review - which part is for contract and which part is for command and why do we overwrite `paramName`.

This is related to the comment [here](https://github.com/keep-network/keep-common/pull/102#discussion_r965723246). If we do this rename, I will consider the other comment resolved.

Reply:
818876bc0307ac17883060a20ca4256e57ddea08